### PR TITLE
[EVM-Equivalent-Yul] Add some environment opcodes

### PR DESCRIPTION
**Pull request by ManuelBilbao**
## What ❔

This PR adds `gas`, `address`, `pc`, `balance` and `origin` opcodes tests

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: core/lib/multivm/src/versions/vm_latest/tests/evm_simulator.rs [hunk 1]
```diff
@@ -1483,6 +1483,176 @@ fn test_basic_jumpi_vectors() {
...
 }
 
+#[test]
+fn test_basic_address_vectors() {
+    assert_eq!(
+        test_evm_vector(
+            vec![
+                // push1 0xFF
+                hex::decode("60").unwrap(),
+                hex::decode("FF").unwrap(),
```
Comment (gianbelinche) [inline core/lib/multivm/src/versions/vm_latest/tests/evm_simulator.rs:1493]: Is this push needed in the test?

File: contracts
Comment (gianbelinche) [inline contracts:1]: Same as the other PR, we should not push this

**Review by gianbelinche — APPROVED** [review]
LGTM

File: core/lib/multivm/src/versions/vm_latest/tests/evm_simulator.rs (hunk 1 above)
Comment (ManuelBilbao) [reply core/lib/multivm/src/versions/vm_latest/tests/evm_simulator.rs:1493]: I put them so if address opcode don't push anything to stack, it don't do weird things

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**